regression/cbmc-incr-smt2/bitvector-bitwise-operators/shift_right.c

```diff
@@ -0,0 +1,24 @@
+int main()
+{
+  // In C, whether a right-shift is arithmetic or logical depends on the
+  // original type being shifted. An unsigned value will be shifted to
+  // the right in a logical manner (this assigns `0` to the leftmost bit).
+  // If the type is signed, right shift will assign the sign bit to the
+  // leftmost digit.
+
+  int bitmask_2 = 0x2;
+  int bitmask_3 = 0x3;
```

Comment:
The comment above is talking about the left most (most significant) bits. But these initialisations are setting the right most (least significant) bits.

Reply:
Yes, the two are unrelated.

The bit masks are there to drive the assertions. The comment is a general comment about how the feature under test behaves and what kind of behaviour we expect to see.

Is this clear, or is there any change you'd prefer to see to make this clearer?

Reply:
Ah, it wasn't clear to me that it was a generalised comment rather than referring to the immediately following line of code. Perhaps it should either live above the main function or in the `.desc` file?

Reply:
Yeah, I'll see to it moved.

Reply:
:+1: